airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/add_fields.py

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+from dataclasses import dataclass
+from typing import Any, List, Mapping, Union
+
+import dpath.util
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.declarative.types import Config, FieldPointer, StreamSlice, StreamState
+
+
+@dataclass(frozen=True)
+class AddedFieldDefinition:
+    path: FieldPointer
+    value: Union[InterpolatedString, str]
```

Comment:
We can probably simplify the code and interface a bit if we change value to just be type `InterpolatedString`. Similar to some of the changes in the retrier code, we can just treat this as a single time coming in

Reply:
Is every input string an interpolated string? Is it safe to assume this will be an interpolated string?

Reply:
the factory will convert the string to InterpolatedString if that's what the type hint is, but I'm not sure this will work well long-term because the json schema will complain that the string is not the object it expects (unless we do some voodoo to replace InterpolatedString with Union[InterpolatedString, str] in the schema).

Reply:
keeping it as union per discussion